firestore/pipeline_result.go

```diff
@@ -88,25 +88,23 @@ func newPipelineResult(ref *DocumentRef, proto *pb.Document, c *Client, executio
 //
 //	var m map[string]interface{}
 //	p.DataTo(&m)
-//
-// except that it returns nil if the document does not exist.
-func (p *PipelineResult) Data() map[string]interface{} {
-	if !p.Exists() {
-		return nil
+func (p *PipelineResult) Data() (map[string]interface{}, error) {
+	if p == nil {
+		return nil, status.Errorf(codes.NotFound, "document does not exist")
```

Comment:
Will the data always be related to a document? Some stages aggregate data across documents, but I'm not too sure if that applies here or not